lib/filewatch/tail.rb

```diff
@@ -183,7 +184,12 @@ def _read_file(path, &block)
             yield(path, line)
             @sincedb[@statcache[path]] += (line.bytesize + delimiter_byte_size)
           end
-        rescue Errno::EWOULDBLOCK, Errno::EINTR, EOFError
+        rescue EOFError
```

Comment:
So the way the code works (i think, based on debug statements) is that it loops trying to read the file with sysread(32768), so if file is < 32K once through the loop gets all content and the next time through sysread will raise EOFError, and its only this event (eof) I want to react to.  Maybe `until file.eof do` as a loop would have been better, but I don't want to change too much.